.buildkite/pipeline.elastic-agent-package.yml

```diff
@@ -126,3 +127,38 @@ steps:
         export DRA_VERSION
       fi
       .buildkite/scripts/steps/dra-publish.sh
+
+  - label: "Publishing via BK API for Independent Agent Release"
+    if: build.env("BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG") == "independent-agent-release-staging"
+    key: bk-api-publish-independent-agent
+    depends_on: package
+    agents:
+      provider: "gcp"
+    env:
+      DRA_PROJECT_ID: "elastic-agent-package"
+      DRA_PROJECT_ARTIFACT_ID: "agent-package"
+    command: |
+      echo "+++ Restoring Artifacts"
+      buildkite-agent artifact download "build/**/*" .
+      echo "+++ Changing permissions for the BK API commands"
+      sudo chown -R :1000 build/distributions/
+      echo "--- File listing"
+      ls -alR build
+      echo "--- Copy workaround for ironbank container filename"
+      .buildkite/scripts/steps/ironbank-cp-workaround.sh
+      echo "--- File listing after workaround"
+      ls -alR build
+      echo "+++ Checking artifact validity with release-manager collect dry run"
+      DRA_DRY_RUN="--dry-run"
+      export DRA_DRY_RUN
+      .buildkite/scripts/steps/dra-publish.sh
+      echo "+++ Call buildkite-agent upload to save artifacts for job that triggered us"
```

Comment:
why we want to explicitly call the upload, we normally define the uploaded artifacts in the respected field in the `step` definition.

Reply:
My thought was that we are verifying that the artifacts are all present with the above `--dry-run` of the `dra-publish.sh` script.  The normal DRA step here relies on the Release Manager container to do the proper uploads, so the expected files aren't currently defined in this repo.

We will have code for the Independent Release added to the `unified-release` repo that will verify that each expected file is present, which will happen after the artifacts are downloaded from this build.

If we also do the expected file check here, and if there's a change in which artifacts are expected, then we will have two places to change it instead of one.

With the Buildkite migration of the main Unified Release workflow, I believe there will be something that eventually replaces the Release Manager container to upload/verify the artifacts.  But that's more on David and Alpar's plate since I'm not on the Release Eng team.

Reply:
Why we need explicitly to call the `buildkite-agent artifact upload` is my point. 
We normally set it like this
```yaml
  - label: "Blah Blah"
    command: run_me.sh
    artifact_paths:
      - "build/distributions/*"
```

This way no matter your code inside the step, at the end, buildkite is going to try and upload those artifacts.

Reply:
I will change it to use the `artifact_paths` entry instead of `buildkite-agent artifact upload`